.github/workflows/testing.yml

```diff
@@ -1,10 +1,21 @@
 # automatically runs pytest to test the length conversion program whenever code is pushed to github
 
 name: test-conversion-program
-on: push
+
+on: 
+  push:
+  pull_request:
+    branches:
+      - main
+
 jobs:
   run-test-code:
     runs-on: ubuntu-latest
+    container:
```

Comment:
Can we put the `strategy` section before the `container` section for maintainability? It works correctly as written, but it’s a little strange to use the `matrix` before it’s been declared. (I’m assuming that the workflow engine just always implements the strategy first/early)